rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.em

```diff
@@ -204,49 +220,56 @@ publish__@(spec.base_type.type)(
   }
 }
 
-ROSIDL_TYPESUPPORT_OPENSPLICE_CPP_EXPORT_@(spec.base_type.pkg_name)
+ROSIDL_TYPESUPPORT_OPENSPLICE_CPP_EXPORT_@(package_name)
 void
-convert_dds_message_to_ros(const __dds_msg_type & dds_message, __ros_msg_type & ros_message)
+convert_dds_message_to_ros(
+  const __dds_msg_type_@(message.structure.type.name) & dds_message,
+  __ros_msg_type_@(message.structure.type.name) & ros_message)
 {
-@[if not spec.fields]@
+@[if not message.structure.members]@
   (void)ros_message;
   (void)dds_message;
 @[end if]@
-@[for field in spec.fields]@
-  // field.name @(field.name)
-@[  if field.type.is_array]@
+@[for member in message.structure.members]@
+  // member.name @(member.name)
+@[  if isinstance(member.type, NestedType)]@
   {
-@[    if field.type.array_size and not field.type.is_upper_bound]@
-    size_t size = @(field.type.array_size);
+@[    if isinstance(member.type, Array)]@
+    size_t size = @(member.type.size);
 @[    else]@
-    size_t size = dds_message.@(field.name)_.length();
-    ros_message.@(field.name).resize(size);
+    size_t size = dds_message.@(member.name)_.length();
+    ros_message.@(member.name).resize(size);
 @[    end if]@
     for (DDS::ULong i = 0; i < size; i++) {
-@[    if field.type.type == 'bool']@
-      ros_message.@(field.name)[i] = (dds_message.@(field.name)_[i] != 0);
-@[    elif field.type.is_primitive_type()]@
-      ros_message.@(field.name)[i] =
-        dds_message.@(field.name)_[i];
+@[    if isinstance(member.type.basetype, BaseString)]@
+      ros_message.@(member.name)[i] = dds_message.@(member.name)_[i];
```

Comment:
I am not sure what to replace it with without making it harder to read. Do you think the following is better?

```
if BaseType and boolean:
    ...
elif BaseType or BaseString:
    ...
else:
    ....
```

Reply:
Yeah, that's an improvement. Since only one line is deduplicated I'm find with leaving it as is, especially if CI is comes back green with this version.

Reply:
Done in 2337a698d5ed2d357860e6182fa64ba4ae2d6280.

I will have to re-run after rebasing another round anyway.

Reply:
Done in 0c633e2ad8352256265db3f50dc571e3cad40f7a.

I will have to re-run after rebasing another round anyway.